light-client/src/utils/block_on.rs

```diff
@@ -0,0 +1,29 @@
+use std::{future::Future, time::Duration};
+
+use crate::{components::io::IoError, types::PeerId};
+
+/// Run a future to completion on a new thread, with the given timeout.
+///
+/// This function will block the caller until the given future has completed.
+pub fn block_on<F>(f: F, peer: PeerId, timeout: Option<Duration>) -> Result<F::Output, IoError>
```

Comment:
I don't think we're going to solve the sync/`async` conundrum in the short term (and definitely not in this PR), but to get the two to play nicely together we should try to figure out a simple, low-effort way to spawn a single Tokio runtime in a separate thread to which we can submit `async` tasks from synchronous code.